File: extract_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');
const path = require('path');

async function extract() {
    const dir = process.cwd();
    console.log('Starting extraction with pdf-parse@1.1.1');

    const targetFiles = [
        { pattern: 'GAIO Investor Profile.pdf', dest: 'investor_profile_extracted.txt' },
        { pattern: 'Comprehensive Website Structure Strategy', dest: 'strategy_extracted.txt' }
    ];

    for (const target of targetFiles) {
        const items = fs.readdirSync(dir);
        const found = items.find(f => f.includes(target.pattern));
        if (found) {
            console.log(`Processing: ${found}`);
            const dataBuffer = fs.readFileSync(path.join(dir, found));
            try {
                const data = await pdf(dataBuffer);
                fs.writeFileSync(path.join(dir, target.dest), data.text);
                console.log(`SUCCESS: ${target.dest}`);
            } catch (e) {
                console.error(`Error parsing ${found}:`, e.message);
            }
        }
    }
    console.log('PDF_EXTRACTION_COMPLETE');
}

extract();

File: extract_pdf.py
import pypdf
import sys

def extract_text(pdf_path):
    try:
        reader = pypdf.PdfReader(pdf_path)
        text = ""
        for page in reader.pages:
            t = page.extract_text()
            if t:
                text += t + "\n"
        return text
    except Exception as e:
        return f"Error reading {pdf_path}: {e}"

files = [
    r"c:\Users\NC\Desktop\olympiad\Comprehensive Website Structure Strategy_for_Global AI Olympiad (1).pdf",
    r"c:\Users\NC\Desktop\olympiad\GAIO Investor Profile.pdf"
]

with open("pdf_extracted_text.txt", "w", encoding="utf-8") as f:
    for file in files:
        f.write(f"--- START OF {file} ---\n")
        f.write(extract_text(file))
        f.write(f"\n--- END OF {file} ---\n")
